### PR TITLE
Little refactor - receive is no longer blocking

### PR DESCRIPTION
**Pull request by Dreat**
Send is the main loop, receive is spawned, so it runs in the background,
basically allowing chat feature.

Next step is to be able to communicate both elixir clients with themselves.

### CODE REVIEW

File: lib/zmq_ex.ex [hunk 1]
```diff
@@ -8,36 +8,60 @@ defmodule ZmqEx do
   def start do
     opts = [:binary, active: false]
     {:ok, socket} = :gen_tcp.connect('localhost', 5555, opts)
-    start_recv socket
+    pid = spawn fn -> printer_loop() end
```
Comment (tomekowal) [inline lib/zmq_ex.ex]: We usually know `spawn` returns a pid. It would be good to name it somehow :)

File: lib/zmq_ex.ex [hunk 2]
```diff
@@ -8,36 +8,60 @@ defmodule ZmqEx do
...
+  end
+
+  defp printer_loop do
+    receive do
+      {:rec_message, value} ->
+        IO.puts "GOT MESSAGE!"
+        IO.inspect value
+        IO.puts "======="
+        printer_loop()
+        _ ->
```
Comment (tomekowal) [inline lib/zmq_ex.ex]: Wrong indentation. Run `mix format` on your code before committing.

File: lib/zmq_ex.ex [hunk 3]
```diff
@@ -8,36 +8,60 @@ defmodule ZmqEx do
...
-
-  end
+  defp ready(socket), do: :gen_tcp.send(socket, <<4, 41, 5, "READY", 11, "Socket-Type", 0, 0, 0, 6, "DEALER", 8, "Identity", 0, 0, 0 ,0>>)
 
   def check_ready(socket) do
     {:ok, msg} = :gen_tcp.recv(socket, 0)
     IO.inspect msg
     case msg do
-      <<4, 41, 5, "READY", 11, "Socket-Type", 0, 0, 0, 6, "DEALER", 8, "Identity", data :: binary >> -> :send
+      <<4, 41, 5, "READY", 11, "Socket-Type", 0, 0, 0, 6, "DEALER", 8, "Identity", _data :: binary >> -> :send
```
Comment (tomekowal) [inline lib/zmq_ex.ex]: Return rather `:ok`, `{:error, reason}` than custom atoms.

File: lib/zmq_ex.ex [hunk 4]
```diff
@@ -8,36 +8,60 @@ defmodule ZmqEx do
   def start do
     opts = [:binary, active: false]
     {:ok, socket} = :gen_tcp.connect('localhost', 5555, opts)
-    start_recv socket
+    pid = spawn fn -> printer_loop() end
+    send? = start_connection(socket)
```
[the remark this answers is not in the document]
Comment (Dreat) [reply lib/zmq_ex.ex]: It's boolean now :)

File: lib/zmq_ex.ex [hunk 5]
```diff
@@ -8,36 +8,60 @@ defmodule ZmqEx do
...
+    spawn fn -> rec_loop(socket, pid) end
+    if (send? === :send) do
```
[the remark this answers is not in the document]
Comment (Dreat) [reply lib/zmq_ex.ex]: Fixed, my bad

File: lib/zmq_ex.ex (hunk 1 above)
Comment (Dreat) [reply lib/zmq_ex.ex]: Looks more appropriate now? :)

File: lib/zmq_ex.ex (hunk 2 above)
Comment (Dreat) [reply lib/zmq_ex.ex]: Forgot about this! Done!

File: lib/zmq_ex.ex (hunk 3 above)
Comment (Dreat) [reply lib/zmq_ex.ex]: Looks like boolean is better here

**Pull request merged**